### PR TITLE
Support custom amount for Mirasvit Store Credit

### PR DESCRIPTION
**Pull request by johnmccombs1**

### CODE REVIEW

File: Helper/Discount.php [hunk 1]
```diff
@@ -497,7 +497,7 @@ protected function getMirasvitStoreCreditBalanceAmount($quote)
                         ->getInstance()
                         ->getBalance($quote->getCustomerId(), $quote->getQuoteCurrencyCode());
 
-        $amount = $balance->getAmount();
+        $amount = ((float)$quote->getManualUsedCredit() > 0) ? $quote->getManualUsedCredit() : $balance->getAmount();
```
Comment (daisy1754) [inline Helper/Discount.php]: is `getManualUsedCredit` built in method? If not, when this method doesn't exist what would happen?

Comment (danac-gs) [reply Helper/Discount.php]: @daisy1754 @mccottry getters are magic methods in Magento 2. They don't need to be explicitly defined. NULL is returned if property is not found. Which is safe in this case.

Comment (matt-thomason) [reply Helper/Discount.php]: Can we add a comment to this code summarizing what @mccottry explained? Someone coming in fresh looking at this code might not understand this is expected to be dependency injected from `MirasvitStoreCredit` magically.

Comment (johnmccombs1) [reply Helper/Discount.php]: Thank @danac-gs, @daisy1754 it is a magic method. The code works as expected, and even the manual_used_credit doesn't exist on the quote. Magic methods essentially allow us to pass data around without explicitly defining data on the model. @matt-thomason I'll push a new update for the comment.

**Pull request merged**